23.0.0.11 staging (#3444)
OpenLiberty/blogs: posts/2023-11-14-23.0.0.11.adoc

---
layout: post
title: "New vendor metrics for MicroProfile Metrics in Open Liberty 23.0.0.11"
# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/lauracowen
author_github: https://github.com/lauracowen
seo-title: New vendor metrics for MicroProfile Metrics in Open Liberty 23.0.0.11 - OpenLiberty.io
seo-description: New vendor metrics for MicroProfile Metrics that you can add directly to your dashboards, and a security fix.
blog_description: New vendor metrics for MicroProfile Metrics that you can add directly to your dashboards, and a security fix.
open-graph-image: https://openliberty.io/img/twitter_card.jpg
open-graph-image-alt: Open Liberty Logo
---
= New vendor metrics for MicroProfile Metrics in Open Liberty 23.0.0.11
Laura Cowen <https://github.com/lauracowen>
:imagesdir: /
//Blank line here is necessary before starting the body of the post.

> Open Liberty 23.0.0.11 includes new vendor metrics for MicroProfile Metrics that you can add directly to your dashboards in various monitoring tools without any additional computation on your part.  Also announced in this release, but available since 23.0.0.10, is the ability to include all files in a specified directory in your server configuration. There's also a security fix.

Reviewer: ```suggestion
Open Liberty 23.0.0.11 includes a security fix and new vendor metrics for MicroProfile Metrics that you can add directly to your dashboards in various monitoring tools, without any additional computation on your part.  Also announced in this release, but available since 23.0.0.10, is the ability to include all files in a specified directory in your server configuration. 
```

vendor_servlet_request_elapsedTime_per_request_seconds{servlet=" myapp_servletB",} 0.384967335
vendor_servlet_request_elapsedTime_per_request_seconds{servlet=" myapp_servletC",} 1.333671328`

|REST Elapsed Time per Request
|`# HELP REST_request_elapsedTime_per_request_seconds The recent average elapsed response time per RESTful resource method request. +
# TYPE REST_request_elapsedTime_per_request_seconds gauge REST_request_elapsedTime_per_request_seconds{class=”my.package.MyClass",method=”simpleGet",mp_scope=”vendor"} 0.0061460695`
|`# TYPE vendor_REST_request_elapsedTime_per_request_seconds gauge +
# HELP vendor_REST_request_elapsedTime_per_request_seconds The recent average elapsed response time per RESTful resource method request. 
vendor_REST_request_elapsedTime_per_request_seconds{class=”my.package.MyClass",method=”simpleGet"} 0.0024352581`

|===

The Heap Utilization and CPU Utilization metrics are available when the server is started. The Connection Pool, REST, and Servlet metrics are available if the application contains any of the relevant data sources, REST APIs, or servlets, as is the case with the existing vendor metrics.

> The new vendor metrics are available in the `/metrics` output, by enabling the following Microprofile Metrics 5.0 feature in your `server.xml`:

Reviewer: ```suggestion
The new vendor metrics are available in the `/metrics` output when you enable the following Microprofile Metrics 5.0 feature in your `server.xml`:
```

Reviewer: @pgunapal - do you have to enable 5.0? The heading says 3,4, and 5 and I beleive the new metrics are avaialble going back to 3.0. But not sure if you have to enable 5 or just "3 or later"

Reviewer 2: Good catch. This is the text copied from the beta. I think it should be 3 or later. I've updated the text to say to add the relevant version and made clear that the code snippet is an example of this.

This is important because when running on Kubernetes, mounting secrets as a whole folder is the only way to reflect the change from the secret dynamically in the running pod.

In the `location` attribute of the `include` element of the `server.xml` file, enter the directory that contains your configuration files. For example:

[source,xml]
----
    <include location="./common/"/>
----

After you make the changes, you can see the following output in the log:

[source]
----
[AUDIT   ] CWWKG0028A: Processing included configuration resource: /Users/rickyherget/libertyGit/open-liberty/dev/build.image/wlp/usr/servers/com.ibm.ws.config.include.directory/common/a.xml
[AUDIT   ] CWWKG0028A: Processing included configuration resource: /Users/rickyherget/libertyGit/open-liberty/dev/build.image/wlp/usr/servers/com.ibm.ws.config.include.directory/common/b.xml
[AUDIT   ] CWWKG0028A: Processing included configuration resource: /Users/rickyherget/libertyGit/open-liberty/dev/build.image/wlp/usr/servers/com.ibm.ws.config.include.directory/common/c.xml
----
The files in the directory are processed in alphabetical order and subdirectories are ignored.

> For more information about Liberty configuration includes, see link:/docs/latest/reference/config/include.html[Include configuration docs].

Reviewer: ```suggestion
For more information about Liberty configuration includes, see link:/docs/latest/reference/config/server-configuration-overview.html#include-processing[Include configuration docs].
```
